docs/router.md

```diff
@@ -0,0 +1,204 @@
+# About Routers
+
+`gpii.express.router` is a subclass of [`gpii.express.middleware`](middleware.md) that wraps
+[the  Express `router` object](http://expressjs.com/en/4x/api.html#router), which is simply a container for a collection
+of middleware.  In our case, these are usually our child components that have the [`gpii.express.middleware`](middleware.md)
+grade.
+
+Routers can be nested, and you can even nest an instance of a router within itself.  The `path` variable for a given
+router module is combined with the paths of its parents, and that ultimately determines which URLs a given router will
+be given the chance to handle (see example below).  A `path` of `/` indicates that a router is dealing with the entire
+path.
+
+If multiple routers are registered for the same path, the first one to receive the request "wins".  As with middleware,
+you can control the order in which routers are added to the "chain" using
+[namespaces and priorities](http://docs.fluidproject.org/infusion/development/Priorities.html).
+The `path` option is one of two means by which a router can indicate that its scope is limited.  A router may also
+indicate that it only wishes to respond to particular [HTTP request methods](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html).
+
+
+# Middleware Isolation
+
+One of the key strengths of Express 4.x and higher is that routers can safely use different middleware than any other
+router, or Express itself.  This is incredibly important when working with even the most common third-party modules for
+Express.  There are often modules that must have a particular piece of middleware, and other modules that do not work at
+all if that middleware is available.
+
+All requests that are passed to a child component of a given router will have already been processed by any middleware
+which sits "upstream", i.e. ahead of this router in the overall routing chain.  Within the router, each piece of
+middleware will be given the chance to work with the request in the order determined by the
+[namespaces and priorities](http://docs.fluidproject.org/infusion/development/Priorities.html) of the child components
+(see example).
+
+# A Few Practical Examples
+
+Let's say we have an instance of `gpii.express` that is configured as follows:
+
+```
+gpii.express({
```

Comment:
I would prefer not to advertise this bad practice in the docs - since you know that people will just cut and paste! The argument supplied to gpii.express should be defined as a grade with gpii.express as a parent - and then the construction done as a separate act. This avoids mixing directives to the RES POTENTIA and the RES EXTENSA (explanation available on request)